src/extras/named-register.js

```diff
@@ -42,11 +42,14 @@
 
   const resolve = systemJSPrototype.resolve;
   systemJSPrototype.resolve = function (id, parentURL) {
-    if (id[0] === '/' || id[0] === '.' && (id[1] === '/' || id[1] === '.' && id[2] === '/'))
-      return resolve.call(this, id, parentURL);
-    if (id in this.registerRegistry)
-      return id;
-    return resolve.call(this, id, parentURL);
+    try {
+      // Prefer import map (or other existing) resolution over the registerRegistry
+      return resolve.call(this, id, parentURL) || this.registerRegistry[id];
```

Comment:
`|| this.registerRegistry[id]` doesn't seem like it will return the correct type here? And it can be undefined?

Reply:
I was thinking that it was okay for resolve to return `undefined` to indicate it can't resolve it? Or is that done only through throwing an error? I am familiar with the import-map resolve function but not with other implementations that might exist. If we should throw when it doesn't exist, I'll remove the `||` expression

Reply:
The undefined

`this.registerRegistry[id]` is a promise for an array, which also doesn't seem to be a suitable return value for resolve.

`System.resolve` itself should never return undefined though, and should always either throw or return a valid resolution.

Reply:
👍 I'll update